src/orquestra/sdk/_base/_driver/_ce_runtime.py

```diff
@@ -208,10 +208,19 @@ def get_workflow_run_outputs_non_blocking(
 
         if len(result_ids) == 0:
             wf_run = self.get_workflow_run_status(workflow_run_id)
-            raise exceptions.WorkflowRunNotSucceeded(
-                f"Workflow run `{workflow_run_id}` is in state {wf_run.status.state}",
-                wf_run.status.state,
-            )
+            if wf_run.status.state == State.SUCCEEDED:
+                raise exceptions.WorkflowResultsNotReadyError(
```

Comment:
how you thought about retrying for a brief sec?
It would be natual for the user to implement their code as:
wf.run()
wf.wait_for_complete()
wf.get_results().

This would give the user the following error. Or maybe we should publish succeeded state only when results are actually ready?

Reply:
I think it should with like 1s or so delay so users wont notice, than we can throw an error